pkg/util/log/log.go

```diff
@@ -27,6 +28,76 @@ var (
 	rxTolerantResourceID = regexp.MustCompile(`(?i)^(?:/admin)?/subscriptions/([^/]+)(?:/resourceGroups/([^/]+)(?:/providers/([^/]+)/([^/]+)(?:/([^/]+))?)?)?`)
 )
 
+type logrusWrapper struct {
+	entry *logrus.Entry
+	level int
+}
+
+func (lw *logrusWrapper) Enabled() bool {
+	return lw.level <= int(logrus.GetLevel())
+}
+
+func (lw *logrusWrapper) Error(err error, msg string, keysAndValues ...interface{}) {
+	lw.withKeysAndValues(keysAndValues).Error(msg, " ", err)
+}
+
+func (lw *logrusWrapper) withKeysAndValues(keysAndValues []interface{}) *logrus.Entry {
+	if len(keysAndValues) == 0 {
+		return lw.entry
+	}
+	key := ""
```

Comment:
as above, could also be:
```suggestion
	var key string
```